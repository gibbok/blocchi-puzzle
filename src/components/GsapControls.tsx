import React from 'react';
import { Controls, PlayState } from 'react-gsap';

type Props = Readonly<{
  children?: React.ReactNode;
}>;

export function GsapControls({ children }: Props): JSX.Element {
  return <Controls playState={PlayState.stop}>{children}</Controls>;
}
