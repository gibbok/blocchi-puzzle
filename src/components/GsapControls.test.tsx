import renderer from 'react-test-renderer';
import React from 'react';
import { GsapControls } from './GsapControls';

describe('GsapControls', () => {
  it('should render', () => {
    const tree = renderer.create(
      <GsapControls>
        <div>test</div>
      </GsapControls>
    );
    expect(tree).toMatchSnapshot();
  });
});
