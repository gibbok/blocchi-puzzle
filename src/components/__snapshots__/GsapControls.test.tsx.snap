// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GsapControls should render 1`] = `
<div>
  <div>
    test
  </div>
  <div
    style={
      Object {
        "backgroundColor": "#f0f0f0",
        "border": "1px solid #ccc",
        "fontFamily": "verdana, sans-serif",
        "fontSize": "16px",
        "marginTop": "10px",
        "padding": "10px 10px 0 10px",
        "position": "relative",
        "zIndex": 2,
      }
    }
  >
    <input
      onChange={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      step="0.001"
      style={
        Object {
          "margin": "0",
          "width": "100%",
        }
      }
      type="range"
    />
    <div
      style={
        Object {
          "display": "flex",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
          "margin": "0",
        }
      }
    >
      <div>
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f0f0f0",
              "border": "1px solid #999",
              "cursor": "pointer",
              "margin": "10px 10px 10px 0",
              "padding": "5px",
            }
          }
          type="button"
        >
          Play
        </button>
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f0f0f0",
              "border": "1px solid #999",
              "cursor": "pointer",
              "margin": "10px 10px 10px 0",
              "padding": "5px",
            }
          }
          type="button"
        >
          Reverse
        </button>
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f0f0f0",
              "border": "1px solid #999",
              "cursor": "pointer",
              "margin": "10px 10px 10px 0",
              "padding": "5px",
            }
          }
          type="button"
        >
          Pause
        </button>
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f0f0f0",
              "border": "1px solid #999",
              "cursor": "pointer",
              "margin": "10px 10px 10px 0",
              "padding": "5px",
            }
          }
          type="button"
        >
          Stop
        </button>
      </div>
      <span
        style={
          Object {
            "color": "#999",
            "fontSize": "14px",
            "margin": "10px 0",
          }
        }
      />
    </div>
  </div>
</div>
`;
